### ISSUE TITLE
[bug] Problem with dynamic variables

### ISSUE THREAD

**Issue by jordi-petit** [rochacbruno/dynaconf#481]
**Describe the bug**
The use of dynamic variables only seems to work in the outer level.

**To Reproduce**

1. Having the following folder structure

config.py:
```python3
from dynaconf import Dynaconf

settings = Dynaconf(
    envvar_prefix="DYNACONF",
    settings_files=['settings.yaml'],
)

print(settings.a) # ok (A)
print(settings.b) # ok (A)
print(settings.c.d) # ok (D)
print(settings.h) # ok (D)
print(settings.c.e) # dynaconf.vendor.box.exceptions.BoxKeyError: "'Settings' object has no attribute
print(settings.c.f) # dynaconf.vendor.box.exceptions.BoxKeyError: "'Settings' object has no attribute
print(settings.c.g) # dynaconf.vendor.box.exceptions.BoxKeyError: "'Settings' object has no attribute
```

2. Having the following config files:

settings.yaml:
```yaml
a: "A"
b: "@format {this.a}"
c:
  d: "D"
  e: "@format {this.a}"
  f: "@format {this.c.d}"
  g: "@format {this.d}"
h: "@format {this.c.d}"
```

4. Executing under the following environment

pipenv install dynaconf

**Expected behavior**

I would not expect to have an exception raised on quering `settings.c.e` or `settings.c.f`. I would expect it on `settings.c.g` because `this` refers to the whole config.

**Environment (please complete the following information):**
 - OS: MacOS
 - Dynaconf Version: 3.1.2
 - Frameworks in use: None

**Comment by rochacbruno** [comment]
This has been fixed in https://github.com/rochacbruno/dynaconf/pull/446 and will be included in the next release 3.1.3

**Comment by jordi-petit** [comment]
Fine! Thanks for your work!
